pandas/tests/arithmetic/test_datetime64.py

```diff
@@ -778,6 +778,48 @@ def test_dt64arr_iadd_timedeltalike_scalar(
         rng += two_hours
         tm.assert_equal(rng, expected)
 
+    @pytest.mark.filterwarnings(
+        "ignore:Adding/subtracting object-dtype array to DatetimeArray"
+    )
+    @pytest.mark.parametrize(
+        "dt64arr, offset_arr, expected",
+        [
+            (
+                [
+                    pd.Timestamp("2000-01-01"),
+                    pd.Timestamp("2000-02-01"),
+                    pd.Timestamp("2000-05-01"),
+                ],
+                [
+                    pd.offsets.DateOffset(years=1),
+                    pd.offsets.DateOffset(months=2),
+                    pd.offsets.MonthEnd(),
+                ],
+                [
+                    pd.Timestamp("2001-01-01"),
+                    pd.Timestamp("2000-04-01"),
+                    pd.Timestamp("2000-05-31"),
+                ],
+            )
+        ],
+    )
+    @pytest.mark.parametrize(
+        "constr_dt, constr_offset, constr_expected",
```

Comment:
it looks like this is intended to do work similarly to the `box` fixture.  is there a compelling reason not to use that?